NitroxModel/Discovery/InstallationFinders/EpicGamesInstallationFinder.cs

```diff
@@ -1,37 +1,51 @@
-﻿using System;
+using NitroxModel.Discovery.Models;
+using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Text.RegularExpressions;
 
-namespace NitroxModel.Discovery.InstallationFinders
+namespace NitroxModel.Discovery.InstallationFinders;
+
+public class EpicGamesInstallationFinder : IGameFinder
 {
-    public class EpicGamesInstallationFinder : IFindGameInstallation
+    private readonly Regex installLocationRegex = new("\"InstallLocation\"[^\"]*\"(.*)\"");
+
+    public GameInstallation? FindGame(GameInfo gameInfo, IList<string> errors)
     {
-        private readonly Regex installLocationRegex = new("\"InstallLocation\"[^\"]*\"(.*)\"");
+        string commonAppFolder = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData);
+        string epicGamesManifestsDir = Path.Combine(commonAppFolder, "Epic", "EpicGamesLauncher", "Data", "Manifests");
+        if (!Directory.Exists(epicGamesManifestsDir))
+        {
+            errors.Add("Epic games manifest directory does not exist. Verify that Epic Games Store has been installed.");
+            return null;
+        }
 
-        public string FindGame(IList<string> errors = null)
+        string[] files = Directory.GetFiles(epicGamesManifestsDir, "*.item");
+        foreach (string file in files)
         {
-            string epicGamesManifestsDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), @"Epic\EpicGamesLauncher\Data\Manifests");
-            if (!Directory.Exists(epicGamesManifestsDir))
-            {
-                errors?.Add("Epic games manifest directory does not exist. Verify that Epic Games Store has been installed.");
-                return null;
-            }
+            string fileText = File.ReadAllText(file);
+            Match match = installLocationRegex.Match(fileText);
 
-            string[] files = Directory.GetFiles(epicGamesManifestsDir, "*.item");
-            foreach (string file in files)
+            if (match.Success && match.Value.Contains(gameInfo.Name))
             {
-                string fileText = File.ReadAllText(file);
-                Match match = installLocationRegex.Match(fileText);
-                if (match.Success && match.Value.Contains("Subnautica") && !match.Value.Contains("Below"))
+                string matchedPath = match.Groups[1].Value;
+
+                if (!GameInstallationFinder.HasGameExecutable(matchedPath, gameInfo))
                 {
-                    Log.Debug($"Found Subnautica install path in '{Path.GetFullPath(file)}'. Full pattern match: '{match.Value}'");
-                    return match.Groups[1].Value;
+                    errors.Add($"Found valid installation directory at '{matchedPath}'. But game exe is missing.");
```

Comment:
Would add the exes name here so we can see if it not just triggered on `SubnauticaBelowZero` bc you have removed that check
```suggestion
                    errors.Add($"Found valid installation directory at '{matchedPath}'. But game {gameInfo.ExeName} is missing.");
```

Reply:
Should never trigger on BelowZero, since I removed the hardcoded temporary fix when BZ came out, in favor of gameinfo's exe name check